source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,239 @@
+
+    """
+    Implementation.
+    """
+
+    def _initialize_impl(self):
+        """Initializes the sensor handles and internal buffers.
+
+        This function creates handles and registers the provided data types with the replicator registry to
+        be able to access the data from the sensor. It also initializes the internal buffers to store the data.
+
+        Raises:
+            RuntimeError: If the imu prim is not a RigidBodyPrim
+        """
+        # Initialize parent class
+        super()._initialize_impl()
+        # create simulation view
+        self._physics_sim_view = physx.create_simulation_view(self._backend)
+        self._physics_sim_view.set_subspace_roots("/")
+        # check if the prim at path is a rigid prim
+        prim = sim_utils.find_first_matching_prim(self.cfg.prim_path)
+        if prim is None:
+            raise RuntimeError(f"Failed to find a prim at path expression: {self.cfg.prim_path}")
+        # check if it is a RigidBody Prim
+        if prim.HasAPI(UsdPhysics.RigidBodyAPI):
+            self._view = self._physics_sim_view.create_rigid_body_view(self.cfg.prim_path.replace(".*", "*"))
+        else:
+            raise RuntimeError(f"Failed to find a RigidBodyAPI for the prim paths: {self.cfg.prim_path}")
+
+        # Create internal buffers
+        self._initialize_buffers_impl()
+
+    def _update_buffers_impl(self, env_ids: Sequence[int]):
+        """Fills the buffers of the sensor data."""
+        # check if self._dt is set (this is set in the update function)
+        if not hasattr(self, "_dt"):
+            raise RuntimeError(
+                "The update function must be called before the data buffers are accessed the first time."
+            )
+        # obtain the poses of the sensors
+        pos_w, quat_w = self._view.get_transforms()[env_ids].split([3, 4], dim=-1)
+        quat_w = math_utils.convert_quat(quat_w, to="wxyz")
+
+        # store the poses
+        self._data.pos_w[env_ids] = pos_w + math_utils.quat_rotate(quat_w, self._offset_pos_b)
+        self._data.quat_w[env_ids] = math_utils.quat_mul(quat_w, self._offset_quat_b)
+
+        # obtain the velocities of the link COM
+        lin_vel_w, ang_vel_w = self._view.get_velocities()[env_ids].split([3, 3], dim=-1)
+        # if an offset is present or the COM does not agree with the link origin, the linear velocity has to be
+        # transformed taking the angular velocity into account
+        lin_vel_w += torch.linalg.cross(
+            ang_vel_w, math_utils.quat_rotate(quat_w, self._offset_pos_b - self._com_pos_b[env_ids]), dim=-1
+        )
+
+        # NOTE: currently the physx API generates errors when using small masses, will switch back to it once fixed
+        # obtain the acceleration of the link COM
+        # lin_acc_w, ang_acc_w = self._view.get_accelerations()[env_ids].split([3, 3], dim=-1)
+        # # if an offset is present or the COM does not agree with the link origin, the linear acceleration has to
```

Comment:
Would suggest removing commented out code and make a corresponding issue with the snippets for the team to investgate.